src/v/cluster/tm_stm.h

```diff
@@ -400,6 +386,60 @@ class tm_stm final : public raft::persisted_stm<> {
     replicate_quorum_ack(model::term_id term, model::record_batch&& batch);
 
     model::record_batch serialize_tx(tx_metadata tx);
+
+    void upsert_transaction(tx_metadata);
+
+    fragmented_vector<tx_metadata> get_transactions_list() const;
+
+private:
+    std::chrono::milliseconds _sync_timeout;
+    config::binding<std::chrono::milliseconds> _transactional_id_expiration;
+    chunked_hash_map<model::producer_identity, kafka::transactional_id>
```

Comment:
I think we should also consolidate code that updates _pid_tx_id and _transactions.. the reverse mapping (unfortunately) exists only for one usecase of try_abort but from what I see any time we bump an epoch in _transactions we should also gc _pid_tx_id, so we should consolidate the code for these updates (with a utility method or some such).